x-pack/plugins/apm/public/components/app/transaction_details/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -203,27 +225,39 @@ function getRootTransaction(childrenByParentId: IWaterfallGroup) {
   }
 }
 
-export type IServiceColors = Record<string, string>;
+function getLegends(waterfallItems: IWaterfallItem[]) {
+  const onlySpanItems = waterfallItems.filter(
+    (item) => 'legendValues' in item
```

Comment:
This seems like a subtle (perhaps fragile) way of ensure it's a span. What about this?

```suggestion
    (item) => item.docType === 'span'
```